tutorials/interactive/index.md

```diff
@@ -0,0 +1,814 @@
+so the same widget should manage both.
+
+If you don't already have the starting code for the Lakes example,
+get it from the links in the blue [**Note**](#note) box.
+
+### Contents
+
+* [Stateful and stateless widgets](#stateful-stateless)
+* [Creating a stateful widget](#creating-stateful-widget)
+  * [Step 1: Decide who manages the widget's state](#step-1)
+  * [Step 2: Subclass StatefulWidget](#step-2)
+  * [Step 3: Subclass the State object](#step-3)
+  * [Step 4: Plug the stateful widget into the widget tree](#step-4)
+  * [Problems?](#problems)
+* [Managing state](#managing-state)
+  * [The widget manages its own state](#self-managed)
+  * [The parent manages the widget's state](#parent-managed)
+  * [A mix-and-match approach](#mix-and-match)
+* [Other interactive widgets](#other-interactive-widgets)
+  * [Standard widgets](#standard-widgets)
+  * [Material widgets](#material-widgets)
+* [Resources](#resources)
+
+<a name="stateful-stateless"></a>
+## Stateful and stateless widgets
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* All widgets are either stateful or stateless.
+* _State_ means information about a widget that might change.
+  A checkbox's state, for example, would probably
+  include a boolean indicating whether it's currently checked.
+* If a widget changes based on user interaction, or as the result of a
+  data feed (for example), it's _stateful_.
+* All widgets should be _immutable_, meaning once created they cannot change.
+* Being immutable is fine for a stateless widget, because its appearance
+  never changes.
+* When a stateful widget changes (its slider moved, for example),
+  the widget needs to rebuild in order to update the UI.
+* A _State_ object holds the changeable information for a stateful widget,
+  separating the widget's state from its appearance.
+* The class managing the state object tells the widget to update the UI by
+  calling `setState`.
+* Typically, the state object is managed by the widget itself, the parent
+  widget, or some combination.
+* How the widget is implemented determines who manages its state.
+
+</div>
+
+There are two kinds of widgets, stateful and stateless.
+
+A _stateless_ widget informs the user, or makes the UI look more attractive.
+A stateless widget has no internal state to manage.
+[Icon,](https://docs.flutter.io/flutter/material/Icon-class.html)
+[Image,](https://docs.flutter.io/flutter/widgets/Image-class.html) and
+[Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are
+examples of stateless widgets, which subclass
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic and can have state that lives beyond the
+life of the widget itself. The user can interact with a stateful widget
```

Comment:
I'm not sure what this means.

StatefulWidget subclasses have no lifetime. They are immutable and therefore conceptually live forever.
A control in the element tree that is represented by a StatefulElement has a single State and a series of StatefulWidgets throughout its lifetime.